### ISSUE TITLE
Improve differentiation of object types

### ISSUE THREAD

**Issue by EricRahm** [mozilla/j2me.js#1133]
A large portion of j2me.js memory use is in js-compartment/classes/class(Object), it would be useful to be able to break this out into class types. 

By converting generic objects, ie:
    `var blah = { 'some': stuff};`
to typed objects we could get better insight into where the memory usage is, ie:
    `var Stuff = function() {}`
    `var blah = new Stuff()`

**Comment by EricRahm** [comment]
This will also help when going through GC logs to determine what's holding onto what.

**Comment by mbebenita** [comment]
We can trivially count the number of Java object allocations and estimate the size of these objects, if you think that would help.

**Comment by marco-c** [comment]
@EricRahm is this still a problem? Were you able to break the *js-compartment/classes/class(Object)* out?

**Comment by EricRahm** [comment]
@marco-c Yes this is still an issue as far as I know.

**Comment by marco-c** [comment]
Are maps reported separately?
I'm wondering about code like (in compiled-method-cache.js):
```
var cache = new Map();
```

**Comment by EricRahm** [comment]
Yes a `Map` should be reported separately, but the size of it probably wouldn't be large enough to report (vs the size of the stuff that it references).
